pkg/frontend/querymiddleware/model_extra.go

```diff
@@ -42,63 +43,232 @@ func newEmptyPrometheusResponse() *PrometheusResponse {
 	}
 }
 
+type PrometheusRangeQueryRequest struct {
+	Path    string        `protobuf:"bytes,1,opt,name=path,proto3" json:"path,omitempty"`
+	Start   int64         `protobuf:"varint,2,opt,name=start,proto3" json:"start,omitempty"`
+	End     int64         `protobuf:"varint,3,opt,name=end,proto3" json:"end,omitempty"`
+	Step    int64         `protobuf:"varint,4,opt,name=step,proto3" json:"step,omitempty"`
+	Timeout time.Duration `protobuf:"bytes,5,opt,name=timeout,proto3,stdduration" json:"timeout"`
+	Query   string        `protobuf:"bytes,6,opt,name=query,proto3" json:"query,omitempty"`
+	Options Options       `protobuf:"bytes,7,opt,name=options,proto3" json:"options"`
+	// ID of the request used to correlate downstream requests and responses.
+	ID int64 `protobuf:"varint,8,opt,name=id,proto3" json:"id,omitempty"`
+	// Hints that could be optionally attached to the request to pass down the stack.
+	// These hints can be used to optimize the query execution.
+	Hints *Hints `protobuf:"bytes,9,opt,name=hints,proto3" json:"hints,omitempty"`
+}
+
+func (r *PrometheusRangeQueryRequest) GetPath() string {
+	if r != nil {
```

Comment:
Is anything relying on this behaviour (being able to call `GetPath()` on a `nil` PrometheusRangeQueryRequest)? If not, would be nice to remove it to keep things simpler.

(same applies to all the `Get...` methods below)

Reply:
And are all of these `Get...` methods still required, or do we just need the methods required to satisfy the interfaces? Given the fields are exported, we should be OK to access them directly when the caller is using the concrete type rather than an interface.

Reply:
so I intended to kind of keep the actual changes light here and just focus on moving it out of the proto structure.

I am planning to address the getter and setter changes when I complete the MinTime and MaxTime work from the other PR - I am open to either approach (public fields vs. getters/setters) for whichever provides a better interface.

We need to both protect the invariants _and_ satisfy the interfaces, so I am expecting to stick with the getters/setters approach unless I see a good path with public fields.

I will update here to remove the nil checks though

Reply:
Getters and setters makes sense to me.